### ISSUE TITLE
Silence all warnings

### ISSUE THREAD

**Issue by turion** [turion/rhine#89]
All warnings under a recent GHC version (at least 8.2.*) should be taken care of.

**Comment by turion** [comment]
Not sure... I don't like `-XCPP`. I think I'll rather fix a "main" GHC version (currently 8.2) and aim at no warnings under this version. (Won't turn them into errors)

On a side note, I don't want Rhine to accumulate cruft in order to accommodate for older GHC versions. If I want a new feature, and an older GHC doesn't support it, the older GHC will just be dropped in the next release. `-XCPP` encourages cruft of that sort, in my opinion.

**Comment by ggreif** [comment]
I totally agree. Just wanted to ask. I think adding `-Wno-unused-imports` in the `rhine.cabal` is the way to go. If you nod, I'd prepare a PR.

**Comment by ggreif** [comment]
See PR #95.

**Comment by ggreif** [comment]
With PR #95 now only `rhine-examples` contain (a handful) warnings.

**Comment by turion** [comment]
Actually, I've changed my mind on `-Wno-unused-imports`. I think it encourages cruft as well. I want to stay up to date with GHC developments and changes in my dependencies, so if an import becomes obsolete in the default GHC version, it's removed. If that breaks an older GHC, support for that older version is removed. I've committed that and documented it in `Contributing.md`.

So for the purpose of this issue, only warnings under the GHC version coming from `stack.yaml` count, i.e. those you encounter when building it from scratch with stack.

**Comment by ggreif** [comment]
Relevant: https://www.reddit.com/r/haskell/comments/8va8mc/stop_supporting_older_ghcs
